arviz/stats/diagnostics.py

```diff
@@ -822,6 +830,36 @@ def _mcse_mean(ary):
     return mcse_mean_value
 
 
+def _mcse_func_sbm(ary, func):
+    """Compute the Markov Chain error on an arbitrary function."""
+    ary = np.asarray(ary)
+    if _not_valid(ary, shape_kwargs=dict(min_draws=4, min_chains=1)):
+        return np.nan
+    ess = _ess_mean(ary)
+    func_estimate_sd = _sbm(ary, func)
+    mcse_func_value = func_estimate_sd / np.sqrt(ess)
+    return mcse_func_value
+
+def _sbm(ary, func):
+    """Subsampling bootstrap method.
+
+    References
+    ----------
+    .. [1] Doss, Charles R., et al. "Markov chain Monte Carlo estimation of quantiles."
+       *Electronic Journal of Statistics* 8.2 (2014): 2448-2478.
+       https://doi.org/10.1214/14-EJS957
+
+    """
+    flat_ary = np.ravel(ary)
```

Comment:
Is this sensitive for which order the ravel is done?

Reply:
I think it technically is, but there should be no difference (hopefully) if the model has converged. It is also not clear to me how should multiple chains be handled when implementing this algorithm, I started with this flatten approach but I can test a couple options.

Reply:
@OriolAbril have you tested alternative approaches for handling multiple chains yet?

Reply:
I benchmarked 3 different approaches for estimating the `mcse` of the mean
- `ess`: using `ess` to estimate mcse, for reference
- `sbm_stack_chains`: estimating mcse with SBM by concatenating the chains (as done here)
- `sbm_stack_draws`: estimating mcse with SBM by interleaving chains (i.e. flattening on the other dimension)
- `sbm_separate_chains`: estimate the mcse of each chain with SBM and then sum the variances and divide by `nchains^2`.
- `sbm_shuffle`: flatten the chains and shuffle the draws before running SBM
I performed the same benchmark as in https://avehtari.github.io/rhat_ess/ess_comparison.html, and transforming the chains to target different stationary distributions. Here's the result:
![mcse_methods_rmse](https://user-images.githubusercontent.com/8673634/212783954-146528a2-81ad-468d-9895-b7104891bb5e.png)

In all cases SBM underestimates the MCSE; this is particularly severe when autocorrelation is high and sample sizes are low. `sbm_stack_chains` is consistently better than the alternatives though. I didn't even bother plotting `sbm_shuffle`, since it was apparent pretty quickly that it was far worse than the others.